### PR TITLE
Adding IfcVoxelData and subclasses

### LINKED ISSUE #448: [IFC Tunnel] Add IfcVoxelData
IfcVoxelData : Subtype of IfcProductData (need a merge of #440). 
  Name : OPTIONAL IfcLabel
  ValueType : OPTIONAL IfcLabel

IfcIntegerVoxelValues : subtype of IfcVoxelData
  Values : Array 1:? of IfcInteger
  Unit : OPTIONAL IfcUnit

IfcLogicalVoxelValues : subtype of IfcVoxelData
  Values : array 1:? of IfcLogical

IfcVectorVoxelValues : subtype of IfcVoxelData
  Values : Array 1:? of IfcVector

IfcRealVoxelValues : subtype of IfcVoxelData
  Values : Array 1:? of IfcReal
  Unit : OPTIONAL IfcUnit

IfcLabelVoxelValues : subtype of IfcVoxelData
  Values : Array 1:? of IfcLabel

### PR DESCRIPTION
**Pull request by larswik**
Fixes #448

### CODE REVIEW

**Review by anborr — COMMENTED** [review]
For naming consistency, I propose to use ...Data instead of ..Values for all subtypes of IfcVoxelData.

E.g., IfcIntegerVoxelValues would then become IfcIntegerVoxelData.

**Review by SergejMuhic — APPROVED** [review]
Looks good to me.

**Review by anborr — CHANGES_REQUESTED** [review]
For naming consistency, I would prefer to use IfcVoxelxxxData instead of IfcVoxelxxxValues for all subtypes of IfcVoxelData. E.g., IfcIntegerVoxelValues would then become IfcIntegerVoxelData.

The rest is approved.

**Comment by larswik** [comment]
Should now be ok with regards to comments.

**Review by anborr — APPROVED** [review]

**Comment by larswik** [comment]
An optional Unit : IfcUnit should be added to IfcVectorVoxelData.

**Comment by larswik** [comment]
> An optional Unit : IfcUnit should be added to IfcVectorVoxelData.

Fixed in [e8d311e](https://github.com/bSI-InfraRoom/IFC-Specification/pull/458/commits/e8d311e3a638fa64ea7ce1b4d4b0a184b62f7e28)

File: IFC4x3/Sections/Core data schemas/Schemas/IfcProductExtension/Entities/IfcVectorVoxelData/DocEntity.xml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocEntity xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" Name="IfcVectorVoxelData" UniqueId="f5ddf0db-53ab-4f77-82b6-0b9e1b34004a" BaseDefinition="IfcVoxelData" EntityFlags="32">
+	<Attributes>
+		<DocAttribute Name="Values" UniqueId="502300df-5d55-4d36-bdbb-920ef292e386" DefinedType="IfcVector" AggregationType="2" AggregationLower="1">
+			<Documentation>The values assigned to the voxels. First x, then y and lastly z.</Documentation>
+		</DocAttribute>
+		<DocAttribute Name="Unit" UniqueId="5db08de1-7f00-4e8e-bc9a-4805251143e8" DefinedType="IfcUnit" AttributeFlags="1">
+			<Documentation>An optional unit for the vector values.</Documentation>
```
Comment (SergejMuhic) [inline IFC4x3/Sections/Core data schemas/Schemas/IfcProductExtension/Entities/IfcVectorVoxelData/DocEntity.xml]:
```suggestion
			<Documentation>An optional unit for the vector values that overrides the default Magnitude IfcLengthMeasure.</Documentation>
```

**Review by SergejMuhic — APPROVED** [review]

**Pull request merged**